src/driver/sap/SapQueryRunner.ts

```diff
@@ -1167,13 +1167,52 @@ export class SapQueryRunner extends BaseQueryRunner implements QueryRunner {
                 `Column "${oldTableColumnOrName}" was not found in the "${table.name}" table.`,
             )
 
-        if (
+        // Check if this is a safe length increase for compatible types
+        const isSafeLengthIncrease =
+            oldColumn.type.toLowerCase() === newColumn.type.toLowerCase() &&
+            oldColumn.length != null &&
+            newColumn.length != null &&
+            Number(newColumn.length) > Number(oldColumn.length) &&
+            (newColumn.type.toLowerCase() === "varchar" ||
+                newColumn.type.toLowerCase() === "char" ||
+                newColumn.type.toLowerCase() === "nvarchar" ||
+                newColumn.type.toLowerCase() === "nchar") &&
+            oldColumn.isGenerated === newColumn.isGenerated
```

Comment:
I understand the purpose of the check, but this does not suffice. There are many more properties that need to be checked if we want to have a partial definition, e.g. collation, comment (basically everything that goes into the definition). 

Or we should use `Driver.createFullType` when generating the alter table.